### DISCUSSION TITLE
onNodeClick and useState

### DISCUSSION THREAD

**Discussion by Eperuza** [xyflow/xyflow#2436] [Q&A]
Greetings,

I have some custom nodes, which render and are draggable. I would like to be able to display an editing form to modify their data fields using onNodeClick. I have a piece of state in my component: `const [selectedNode, setSelectedNode] = useState(null);` and onNodeClick, I'd like to set that value to the node clicked. When there is a value in that piece of state, a form should display. My issue is that I cannot get state to update using onNodeClick.

This is the definition of onNodeClick 

```
const onNodeClick = useCallback(
    (event, node) => {
      console.log('click node', node); //this line executed just fine and displayed all of the properties and correct data of my custom node
      setSelectedNode(node); //When I set a breakpoint, it was reached, but nothing happened, and state was not updated.
    }, //expected behavior is that the node logged is set to the selectedNode state.
    [setSelectedNode]
  );
```

This is my return function 
```
return (
      <div className = "Modeling">
        <button onClick = {() => handleLoadModel()}>Load Model</button>
        <ReactFlow
        nodeTypes={nodeTypes}
        nodes={nodes}
        edges={edges}
        onNodesChange={onNodesChange}
        onEdgesChange={onEdgesChange}
        fitView
        onNodeClick={onNodeClick}
        />
        {selectedNode && <NodeEditor nodeToEdit = {selectedNode} nodes = {nodes} setNodes = {setNodes} close {setSelectedNode(null)}/>} //this never renders because selectedNode stays null after onClick
      </div>
  );
}
```

this is the code for my custom node:

```
export default function TaskNode({data}) {
  return (
    <div className = {data.dataChanged ? 'task-node dataChanged' : 'task-node'}>
        <Handle type="target" position={Position.Top} />
            {data.label}
        <Handle type="source" position={Position.Bottom}/>
    </div>
  )
}
```

Thank you in advance for any assistance, I hope that I have provided sufficient information.

The rendered nodes (Had to redact the labels) 
![modeling_redacted_ss](https://user-images.githubusercontent.com/79524393/191114685-310da77d-278b-45e3-9638-b938433da24d.png)

**Comment by moklick** [comment]
Hi! Could you create a codesandbox for this?

**Reply by Eperuza** [reply]
Please see the sandbox https://codesandbox.io/s/prod-water-nfqxvj

I had to just put hard coded json for the initial data

**Reply by Eperuza** [reply]
I appear to have managed to fix it thanks to the sandbox... Still would appreciate a quick check if the implementation is correct.
